Device/ARM/ARMCR52/startup_ARMCR52.c

```diff
@@ -0,0 +1,159 @@
+/******************************************************************************
+ * @file     startup_ARMCR52.c
+ * @brief    CMSIS Core Device Startup File for Cortex-R52 Device
+ * @version
+ * @date
+ ******************************************************************************/
+/*
+ * Copyright (c) 2019 Arm Limited. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if defined (ARMCR52)
+  #include "ARMCR52.h"
+#else
+  #error device not specified!
+#endif
+
+/*----------------------------------------------------------------------------
+  Linker generated Symbols
+ *----------------------------------------------------------------------------*/
+extern uint32_t __HypStackTop;
+extern uint32_t __ecc_init_start__, __ecc_init_end__;
```

Comment:
These symbols look GCC-specific to me. We need to sort the compiler specifics out if we want to have the startup.c file agnostic right from the beginning. Another feasible solution would be adding FCC specific startup code, placing it to the GCC subfolder.

Reply:
ECC should be done no matter which compiler is used. I see only way to declare new symbols like __ECC_INIT_START and __ECC_INIT_END and define them in compiler dependent header.